chain/x/mint/internal/keeper/keeper.go

```diff
@@ -111,3 +137,24 @@ func (k Keeper) MintCoins(ctx sdk.Context, newCoins sdk.Coins) error {
 func (k Keeper) AddCollectedFees(ctx sdk.Context, fees sdk.Coins) error {
 	return k.supplyKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.feeCollectorName, fees)
 }
+
+// IsEligibleAccount checks if acc is eligible to mint
+func (k Keeper) IsEligibleAccount(ctx sdk.Context, acc sdk.AccAddress) bool {
+	mintPool := k.GetMintPool(ctx)
+	return mintPool.EligiblePool.Contains(acc)
+}
+
+// IsExceedsLimit checks if minting amount exceeds the limit
+func (k Keeper) IsExceedsLimit(ctx sdk.Context, amt sdk.Coins) bool {
+	moduleParams := k.GetParams(ctx)
+	return amt.IsAnyGT(moduleParams.MintMax)
+}
+
+// MintCoinsToAcc mints coins from module to account
+func (k Keeper) MintCoinsToAcc(ctx sdk.Context, recipient sdk.AccAddress, amt sdk.Coins) error {
+	if amt.Empty() {
+		return nil
+	}
+
+	return k.supplyKeeper.MintCoinsToAcc(ctx, types.ModuleName, recipient, amt)
```

Comment:
You don't need to perform this logic in supply module, do it in minting here, though you won't need to make aliases and export Msg